### PR TITLE
Use ModelBackedDrawable for UpdateableBeatmapSetCover

### LINKED ISSUE #5246: UpdateableBeatmapSetCover should use ModelBackedDrawable

### PR DESCRIPTION
**Pull request by frenzibyte**
Simplifies the class, *the missing one out of the rest*

- [ ] Depends on ppy/osu-framework#2890

Closes #5246

### CODE REVIEW

**Comment by peppy** [comment]
This changes the behaviour. Previously these drawables used to have a background before/while they were loading. Now they are transparent. Please create tests so you can ensure no regressions.

File: osu.Game/Beatmaps/Drawables/UpdateableBeatmapBackgroundSprite.cs
Comment (peppy) [inline osu.Game/Beatmaps/Drawables/UpdateableBeatmapBackgroundSprite.cs]: please do this in its own PR, it's not related.

**Comment by peppy** [comment]
Still behaving differently - see osu!direct panels

**Comment by iiSaLMaN** [comment]
> Still behaving differently - see osu!direct panels

I'm not sure how to fix this, the issue is because MBD doesn't have the chance to draw the placeholder box

File: osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs
Comment (swoolcock) [inline osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs]: Why are `new`-ing this?  I don't see the need for this redeclaration when `base.Masking` is public, and you're not doing anything more than calling `base`.

File: osu.Game.Tests/Visual/Online/TestSceneUpdateableBeatmapSetCover.cs
Comment (swoolcock) [inline osu.Game.Tests/Visual/Online/TestSceneUpdateableBeatmapSetCover.cs]: Would be nice to have some asserts here to ensure the `BeatmapSetInfo` has been correctly set for each `BeatmapSetCover`.

**Review by swoolcock — CHANGES_REQUESTED** [review]

File: osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs
Comment (peppy) [reply osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs]: this is resharper enforced, for clarity when working with the class itself (sometimes you intentionally want to call `base.`).

Comment (swoolcock) [reply osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs]: My point was more why does the property exist at all.

Comment (peppy) [reply osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs]: `ModelBackedDrawable` is a composite, which doesn't expose masking itself. That said, you may have a point – could we not just set `Masking` via protected in all use cases? Seems saner to me.

Comment (swoolcock) [reply osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs]: `CompositeDrawable.Masking` is public.

Comment (iiSaLMaN) [reply osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs]: actually, i think it would be better if i did the masking inside `BeatmapSetCover`

File: osu.Game.Tests/Visual/Online/TestSceneUpdateableBeatmapSetCover.cs
Comment (iiSaLMaN) [reply osu.Game.Tests/Visual/Online/TestSceneUpdateableBeatmapSetCover.cs]: should it ensure on null beatmapset too?

**Comment by iiSaLMaN** [comment]
Does not require an issue fix anymore, ready for review.

**Comment by iiSaLMaN** [comment]
I'm not sure whether to wait for ppy/osu-framework#2634 to be fixed and create the placeholder box inside `CreateDrawable`, or separate that into another PR (if this gets merged)

**Comment by smoogipoo** [comment]
image

Masking seems to be broken, and is a framework-side issue with nested masking that I can't find right now. Why is it being masked in the first place?

**Comment by iiSaLMaN** [comment]
> image
> 
> Corner masking seems to be broken, and is a framework-side issue with nested masking that I can't find right now. Why is it being masked in the first place?

**Comment by smoogipoo** [comment]
Where? It seems like all the sizes are the full size of the contained?

**Comment by iiSaLMaN** [comment]
> Where? It seems like all the sizes are the full size of the container?

`BeatmapSetCover` sprite is bigger than the container itself, although it shouldn't be masked for the panel, will revert a commit I made, should fix this

**Comment by iiSaLMaN** [comment]
I've applied major refactoring to this, should be ready to merge after framework-side bug fix is merged.

**Comment by frenzibyte** [comment]
Closing as way too old and superseded by #11582

**Pull request closed without merging**
